intern.go

```diff
@@ -86,6 +86,33 @@ func Get(cmpVal interface{}) *Value {
 	return v
 }
 
+// GetByString is identical to Get, except that it is specialized for strings.
+// This avoids an allocation from putting a string into an interface{}
+// to pass as an argument to Get.
+func GetByString(s string) *Value {
+	mu.Lock()
+	defer mu.Unlock()
+
+	var v *Value
+	if valSafe != nil {
+		v = valSafe[s]
+	} else if addr, ok := valMap[s]; ok {
```

Comment:
How about making a new `type key struct { s string, ei interface{} }` type, and make the maps keyed by that instead. Then Get and GetString can both call an unexported `func get` that takes that wide key (we could even add an int64 to it later).

It's pretty sad to have that gnarly subtle code duplicated.